packages/devtools_app/lib/src/shared/routing.dart

```diff
@@ -60,7 +60,9 @@ class DevToolsRouteInformationParser
       // If the uri has been modified and we do not have a vm service uri as a
       // query parameter, ensure we manually disconnect from any previously
       // connected applications.
-      await serviceConnection.serviceManager.manuallyDisconnect();
+      if (serviceConnection.serviceManager.hasConnection) {
+        await serviceConnection.serviceManager.manuallyDisconnect();
+      }
```

Comment:
I'm not too familiar with this, but is it possible we'd have a race here if there is a connection in progress (`hasConnection` being false so we skip this, but a connection ends up being formed shortly after?)

(if it is possible, I don't know if calling `manuallyDisconnect()` is the right thing to do to cancel that anyway, so it might be an unrelated issue if there is such an issue)